beginner_source/torchtext_translation.py

```diff
@@ -40,49 +40,41 @@
 
 import torchtext
 import torch
+# With torchtext 0.9.0 rc
+# from torchtext.datasets import Multi30k
+from torchtext.experimental.datasets.raw import Multi30k
 from torchtext.data.utils import get_tokenizer
 from collections import Counter
 from torchtext.vocab import Vocab
-from torchtext.utils import download_from_url, extract_archive
-import io
-
-url_base = 'https://raw.githubusercontent.com/multi30k/dataset/master/data/task1/raw/'
-train_urls = ('train.de.gz', 'train.en.gz')
-val_urls = ('val.de.gz', 'val.en.gz')
-test_urls = ('test_2016_flickr.de.gz', 'test_2016_flickr.en.gz')
-
-train_filepaths = [extract_archive(download_from_url(url_base + url))[0] for url in train_urls]
-val_filepaths = [extract_archive(download_from_url(url_base + url))[0] for url in val_urls]
-test_filepaths = [extract_archive(download_from_url(url_base + url))[0] for url in test_urls]
-
-de_tokenizer = get_tokenizer('spacy', language='de')
-en_tokenizer = get_tokenizer('spacy', language='en')
-
-def build_vocab(filepath, tokenizer):
-  counter = Counter()
-  with io.open(filepath, encoding="utf8") as f:
-    for string_ in f:
-      counter.update(tokenizer(string_))
-  return Vocab(counter, specials=['<unk>', '<pad>', '<bos>', '<eos>'])
-
-de_vocab = build_vocab(train_filepaths[0], de_tokenizer)
-en_vocab = build_vocab(train_filepaths[1], en_tokenizer)
-
-def data_process(filepaths):
-  raw_de_iter = iter(io.open(filepaths[0], encoding="utf8"))
-  raw_en_iter = iter(io.open(filepaths[1], encoding="utf8"))
+
+train_iter, = Multi30k(split=('train'))
```

Comment:
We should add support for a single string for this release as well, i.e.
```
train_iter = Multi30k(split='train')
```
I'll send a PR to torchtext.

Reply:
We now have support for a single string argument. I'd prefer if we used that one.